spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.example;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.gcp.pubsub.core.PubSubOperations;
+import org.springframework.cloud.gcp.pubsub.support.GcpHeaders;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.gcp.pubsub.AckMode;
+import org.springframework.integration.gcp.pubsub.inbound.PubSubInboundChannelAdapter;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class WebController {
```

Comment:
Why do you need a web controller, if you're not actually serving anything? In fact, you don't need the port or the web starter.

Reply:
Agree. This application does nothing with Web. Let's get rid of it here!